lib/internal/bootstrap/node.js

```diff
@@ -625,5 +627,59 @@
     new vm.Script(source, { displayErrors: true, filename });
   }
 
+  function setupAllowedFlags() {
+    // This builds process.allowedNodeEnvironmentFlags
+    // from data in the config binding
+
+    const replaceDashesRegex = /-/g;
+    const leadingDashesRegex = /^--?/;
+    const trailingValuesRegex = /=.*$/;
+
+    const trimLeadingDashes = (flag) => flag.replace(leadingDashesRegex, '');
+
+    const {
+      allowedV8EnvironmentFlags,
+      allowedNodeEnvironmentFlags
+    } = process.binding('config');
+
+    // for comparison purposes only
+    const v8Flags = new Set(allowedV8EnvironmentFlags
+      .map(trimLeadingDashes));
+    const nodeFlags = new Set(allowedNodeEnvironmentFlags
+      .map(trimLeadingDashes));
+
+    class NodeEnvironmentFlagsSet extends Set {
+      constructor(...args) {
+        // There is no way to actually freeze or otherwise
+        // make a Set immutable, AFAIK.  Consider this
+        // "best effort" using "minimal effort"
+        super(...args);
+
+        this.add = () => this;
+        this.clear = () => {};
+        this.delete = () => false;
+        Object.freeze(this);
+      }
+
+      has(key) {
+        if (typeof key === 'string') {
+          key = key.replace(trailingValuesRegex, '');
+          if (leadingDashesRegex.test(key)) {
+            return Set.prototype.has.call(this, key) ||
```

Comment:
this isn’t robust against deletion of the has method; freezing the prototype of your subclass would mean you could reliably do this.has

Reply:
that doesn’t make sense b/c I’m overriding `has`? but regardless I can do the same as the previous comments.

Reply:
you're *shadowing* it, but you still call into the builtin one here.